### PR TITLE
SWC-5435: do not retry requests that result in 500 errors, and throw …

### PR DESCRIPTION
**Pull request by jay-hodgson**
…the error instead of returning the error object.

### CODE REVIEW

File: src/lib/containers/download_list/CreatePackage.tsx [hunk 1]
```diff
@@ -78,11 +78,11 @@ export const CreatePackage = (props: CreatePackageProps) => {
         zipFileName: fileNameWithZipExtension,
         zipFileFormat: ZipFileFormat.Flat,
       }
-      const bulkFileDownloadResponse: BulkFileDownloadResponse = await getBulkFiles(
+      const currentBulkFileDownloadResponse: BulkFileDownloadResponse = await getBulkFiles(
```
Comment (jay-hodgson) [inline src/lib/containers/download_list/CreatePackage.tsx:81]: rename to fix scope warning

File: src/lib/containers/download_list/DownloadListTable.tsx [hunk 2]
```diff
@@ -125,31 +125,31 @@ export default function DownloadListTable(props: DownloadListTableProps) {
     }
     try {
       setIsLoading(true)
-      const downloadList = await getDownloadList(token)
-      const { filesToDownload } = downloadList
-      if (filesToDownload.length === 0) {
+      const currentDownloadList: DownloadList = await getDownloadList(token)
```
Comment (jay-hodgson) [inline src/lib/containers/download_list/DownloadListTable.tsx:128]: rename to fix scope warning (many times in this file)

File: src/lib/utils/SynapseClient.ts [hunk 3]
```diff
@@ -136,10 +136,12 @@ export type SynapseClientError = {
...
   0 - no internet connection
-  429 - too many concurrent requests
-  500>= - any status code of 500 or more is a server side error
+  429 - Too Many Requests
+  502 - Bad Gateway
+  503 - Service Unavailable
+  504 - Gateway Timeout
 */
-const RETRY_STATUS_CODES = [0, 429, 500, 502, 503, 504]
+const RETRY_STATUS_CODES = [0, 429, 502, 503, 504]
```
Comment (jay-hodgson) [inline src/lib/utils/SynapseClient.ts:144]: Do not retry general 500 (Internal Server Error) responses, just throw the error.

**Review by nickgros — APPROVED** [review]
Looks good to me!

**Pull request merged**